### PR TITLE
UPD: updates network checks for multiphase networks

### PR DESCRIPTION
**Pull request by pseudocubic**
Updates the following functions to be compatible with multiphase networks

* `check_thermal_limits`
* `check_branch_directions`
* `check_transformer_parameters`
* `check_dcline_limits`
* `check_voltage_setpoints`
* `check_cost_functions`

These updates only handle diagonal cases, off-diagonal cases should be handled
in ThreePhasePowerModels (e.g. in `check_thermal_limits`).

Other checks that were not updated were confirmed to work on Multiphase networks.

Added `Base.:*` and `Base.:/` extensions for MultiPhaseVectors (handles
special broadcast cases).

Fixed issue with `Base.broadcast` extensions.

Added unit tests for new features.

### CODE REVIEW

File: src/core/data.jl [hunk 1]
```diff
@@ -788,33 +806,35 @@ end
...
-            end
-            if length(comp["cost"]) < 4
-                error("cost includes $(comp["ncost"]) points, but at least two points are required")
-            end
-            for i in 3:2:length(comp["cost"])
-                if comp["cost"][i-2] >= comp["cost"][i]
-                    error("non-increasing x values in pwl cost model")
+        for ph in 1:length(comp["ncost"])
+            if comp["model"][ph] == 1
+                if length(PowerModels.getmpv(comp["cost"], ph)) != 2*comp["ncost"][ph]
```
Comment (ccoffrin) [inline src/core/data.jl]: Note, inside of PowerModels you can call `getmpv` without needing to explicitly quantify it.

**Comment by pseudocubic** [comment]
@ccoffrin I had an idea about this problem with the warning messages last night, see my latest commit.

File: src/core/data.jl [hunk 2]
```diff
@@ -386,6 +386,7 @@ function _rescale_cost_model(comp::Dict{String,Any}, scale::Real, multiphase::Bo
         else
             phases = length(comp["model"])
             for ph in 1:phases
+                ph_str = phases > 1 ? " on phase $(ph)" : ""
```
Comment (ccoffrin) [inline src/core/data.jl]: Seems reasonable.  Consider putting the string generation into a function to try to enforce consistency.

Note that _implicit single phase_ occurs when the "phases" term is omitted from the data.  If the "phases" term is set to 1, then it is _explicit single phase_ and it is still ok to print "on phase 1".

Comment (pseudocubic) [reply src/core/data.jl]: I did think about a function, but the wording of the warnings were not conducive to always using the same string, therefore making a function wouldn't save much duplication.

In that case, I will use `haskey(data, "phases")` in most cases. In the case you quoted above, `data` isn't passed to the function, so I will just check the type of `comp["model"]`

**Pull request merged**